app/controllers/reviews_controller.rb

```diff
@@ -5,6 +5,11 @@ def show
 
   def update
     @review = Review.find(params[:id])
+    if current_user.nil? || @review.user_id != current_user.id
+      flash[:notice] = "You must log in to edit a review"
+      redirect_to station_path(@review.station)
+      return
+    end
```

Comment:
you don't want to be checking if a user is present in your controllers, extract that logic into models and views